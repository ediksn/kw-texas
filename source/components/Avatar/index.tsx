import React, { useState } from 'react'
import { Image, ImageStyle, Text, TextStyle, View, ViewStyle } from 'react-native'
import DefaultAvatar from 'assets/images/default-avatar.png'
import { useSelector } from 'react-redux'
import { styles } from './styles'
import { useGetInitials } from '~/hooks'
import { ProfileInterface } from '~/interfaces/usrInterface'
import { RootState } from '~/store'

interface Props {
  uri?: any
  testID?: string
  avatarStyle?: ImageStyle
  avatarDefaultStyle?: ImageStyle
  initialsViewStyle?: ViewStyle
  initialsStyle?: TextStyle
  forceInitials?: boolean
}

const Avatar = ({
  uri,
  testID,
  avatarStyle,
  avatarDefaultStyle,
  initialsViewStyle,
  initialsStyle,
  forceInitials = false
}: Props) => {
  const profiles: ProfileInterface[] = useSelector((state: RootState) => state.usrProfile.profiles)
  const activeAccount: number = useSelector((state: RootState) => state.usrProfile.activeAccount)
  const currentProfile = (profiles && !!profiles.length && profiles[activeAccount]) || null
  const [imageError, setImageError] = useState(false)
  const uriOk = typeof uri !== 'undefined' && uri !== ''
  const imageOk = !imageError && uriOk

  return imageOk ? (
    <Image
      testID={testID}
      resizeMode='cover'
      source={{ uri }}
      style={[styles.avatar, avatarStyle]}
      onError={() => !imageError && setImageError(true)}
    />
  ) : !imageOk && forceInitials ? (
    <View style={[styles.initialsView, initialsViewStyle]}>
      <Text style={[styles.initials, initialsStyle]}>{useGetInitials(currentProfile?.name)}</Text>
    </View>
  ) : (
    <Image
      testID={testID}
      resizeMode='cover'
      source={DefaultAvatar}
      style={[styles.avatarDefault, avatarDefaultStyle]}
    />
  )
}

export default Avatar
