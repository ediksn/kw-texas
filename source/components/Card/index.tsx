import React from 'react'
import { View, Text, Image, TouchableOpacity } from 'react-native'
import { styles } from './styles'
import mockPost from './__mocks__/mockPost'
import avatarMock from './__mocks__/avatarNinaRoyals.png'
import threeDotsMock from './__mocks__/threeDots.png'

const Card = () => {
  const { author, date } = mockPost

  const Header = () => (
    <View style={styles.header}>
      <Image style={styles.avatar} resizeMode='center' source={avatarMock} />
      <View style={styles.info}>
        <Text style={styles.name}>{author.name.toUpperCase()}</Text>
        <Text>{date}</Text>
      </View>
      <TouchableOpacity onPress={() => null}>
        <Image style={styles.threeDots} resizeMode='center' source={threeDotsMock} />
      </TouchableOpacity>
    </View>
  )

  return (
    <View style={styles.container}>
      <Header />
    </View>
  )
}

export default Card
