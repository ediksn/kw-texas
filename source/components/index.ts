/* eslint-disable import/no-cycle */
export { default as Avatar } from './Avatar'
export { default as Button } from './Button'
export { default as CheckBox } from './CheckBox'
export { default as Header } from './Header'
export { default as Input } from './Input'
export { default as Spinner } from './Spinner'
export { default as Video } from './NativeVideo'
export { default as VideoCard } from './VideoCard'
export { default as VideoList } from './VideoList'
export { default as HeaderTitleBox } from './HeaderTitleBox'
export { default as Post } from './Post'
export { default as Icon } from './Icon'
export { default as Dropdown } from './Dropdown'
export { default as Toast } from './Toast'
export { default as EventCard } from './EventCard'
export { default as EmptyList } from './EmptyList'
export { default as EventsCarrousel } from './Carrousel'
export { default as PostInfo } from './PostInfo'
export { default as PostMedia } from './PostMedia'
export { default as PostLinksForm } from './PostLinksForm'
export { default as TagsPeopleHeader } from './TagsPeopleHeader'
export { default as Tags } from './Tags'
