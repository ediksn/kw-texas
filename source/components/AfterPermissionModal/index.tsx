import React from 'react'
import { useTranslation } from 'react-i18next'
import { Linking, Platform, Text, View } from 'react-native'
import Modal from 'react-native-modal'
import Button from '../Button'
import { styles } from './styles'

interface Props {
  isVisible: boolean
  biometryType: string
  handleCloseModal: () => void
}
const AfterPermissionModal = ({ isVisible, biometryType, handleCloseModal }: Props) => {
  const { t } = useTranslation()

  const handleGoToSettings = () => {
    handleCloseModal()
    if (Platform.OS === 'ios') {
      Linking.openURL('app-settings://permissions/myapp')
    } else {
      Linking.openSettings()
    }
  }
  return (
    <Modal isVisible={isVisible} onBackdropPress={() => null}>
      <View style={styles.container}>
        <Text style={styles.titleText}>{t(`components_AfterPermissionModal_TurnOn${biometryType}`)}</Text>
        <Text style={styles.bodyText}>{t(`components_AfterPermissionModal_Opportunity${biometryType}`)}</Text>
        <View style={styles.buttonsContainer}>
          <Button
            viewStyle={styles.button}
            message={t('components_AfterPermissionModal_GoToSettings')}
            type='TEXT'
            onPress={handleGoToSettings}
          />
          <Button viewStyle={styles.button} message={t('general_text_Ok')} onPress={handleCloseModal} />
        </View>
      </View>
    </Modal>
  )
}

export default AfterPermissionModal
