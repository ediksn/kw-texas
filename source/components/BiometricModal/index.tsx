/** @format */

import React from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TouchableOpacity, View } from 'react-native'
import ReactNativeBiometrics from 'react-native-biometrics'

import { Icon } from '~/components'
import { theme } from '~/constants'

import { styles } from './styles'

interface Props {
  onAuth: any
  biometryType: string
  setBiometricPosition: React.Dispatch<React.SetStateAction<number>>
}

const BiometricModal = ({ onAuth, biometryType, setBiometricPosition }: Props) => {
  const { t } = useTranslation()

  const texts: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('components_BiometricModal_FaceID'),
    [ReactNativeBiometrics.Biometrics]: t('components_BiometricModal_Fingerprint'),
    [ReactNativeBiometrics.TouchID]: t('components_BiometricModal_Fingerprint')
  }
  const icons: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: 'faceid-icon',
    [ReactNativeBiometrics.TouchID]: 'fingerprint-icon',
    [ReactNativeBiometrics.Biometrics]: 'fingerprint-icon'
  }

  return (
    <View
      style={styles.container}
      onLayout={event => {
        setBiometricPosition(event.nativeEvent.layout.y)
      }}
    >
      <Text style={styles.text}>{texts[biometryType]}</Text>
      <TouchableOpacity onPress={onAuth}>
        <Icon size={40} name={icons[biometryType]} color={theme.activeColor} />
      </TouchableOpacity>
    </View>
  )
}

export default BiometricModal
