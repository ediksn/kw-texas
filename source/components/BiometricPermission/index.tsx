/** @format */

import React, { memo } from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TouchableOpacity, View } from 'react-native'
import ReactNativeBiometrics from 'react-native-biometrics'

import { ModalBottom } from '~/components/ModalBottom'

import { styles } from './styles'

interface Props {
  onYes: () => void
  onNo: () => void
  onRequestClose: () => void
  biometryType: string
  isVisible: boolean
}

const BiometricPermission = memo(({ onYes, onNo, onRequestClose, biometryType, isVisible }: Props) => {
  const { t } = useTranslation()

  const headers: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('Face ID'),
    [ReactNativeBiometrics.TouchID]: t('Fingerprint'),
    [ReactNativeBiometrics.Biometrics]: t('Fingerprint')
  }
  const bodies: { [index: string]: string } = {
    [ReactNativeBiometrics.FaceID]: t('Do you want to allow app to use Face ID to authenticate?'),
    [ReactNativeBiometrics.TouchID]: t('Do you want to allow app to use Fingerprint to authenticate?'),
    [ReactNativeBiometrics.Biometrics]: t('Do you want to allow app to use Fingerprint to authenticate?')
  }

  return (
    <ModalBottom visible={isVisible} onClose={onRequestClose}>
      <Text style={styles.header}>{headers[biometryType]}</Text>

      <Text style={styles.message}>{bodies[biometryType]}</Text>

      <View style={styles.buttons}>
        <TouchableOpacity onPress={onNo} style={[styles.button, styles.buttonNegative]}>
          <Text style={styles.textNegative}>{t('No')}</Text>
        </TouchableOpacity>

        <TouchableOpacity onPress={onYes} style={styles.button}>
          <Text style={styles.textPositive}>{t('Yes')}</Text>
        </TouchableOpacity>
      </View>
    </ModalBottom>
  )
})

export default BiometricPermission
