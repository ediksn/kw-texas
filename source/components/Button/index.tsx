import React, { memo } from 'react'
import { useTranslation } from 'react-i18next'
import { Text, TextStyle, TouchableOpacity, View, ViewStyle } from 'react-native'
import { moderateScale } from 'react-native-size-matters'
import { styles, stylesOfType } from './styles'
import { theme } from '~/constants'
import { Icon } from '~/components'

interface Props {
  testID?: string
  message?: string
  THEME?: any
  fontSize?: number
  type?: string
  viewStyle?: ViewStyle
  textStyle?: TextStyle
  disabled?: boolean
  onPress?: () => void
  icon?: any
}

const Button = ({
  testID,
  message = 'Ok',
  THEME = theme.buttons.primary,
  fontSize = moderateScale(theme.fonts.MEDIUM_SIZE),
  type = theme.buttons.types.CONTAINED,
  viewStyle,
  textStyle,
  disabled,
  onPress,
  icon = { name: '', size: 20, color: '#000000' }
}: Props) => {
  const { backgroundTypeStyle, textTypeStyle } = stylesOfType[type](THEME)

  const Message = () => {
    const { t } = useTranslation()

    return (
      <View style={[styles.messageView, icon && styles.messageWithIconView]}>
        {icon && (
          <View style={styles.icon}>
            <Icon name={icon.name} size={icon.size} color={icon.color} />
          </View>
        )}
        <Text style={[styles.messageText, { color: textStyle?.color }, textTypeStyle, { fontSize }]}>{t(message)}</Text>
      </View>
    )
  }

  return (
    <TouchableOpacity
      testID={testID}
      activeOpacity={0.6}
      disabled={disabled}
      style={[styles.containerView, viewStyle, backgroundTypeStyle]}
      onPress={onPress}
    >
      <Message />
    </TouchableOpacity>
  )
}

export default memo(Button)
