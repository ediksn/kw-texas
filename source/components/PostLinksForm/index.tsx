import React, { Dispatch, SetStateAction } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import { styles } from './styles'
import { Icon, Input } from '~/components'
import { theme } from '~/constants'
import { addToArray, replaceValueFromIndex, removeFromArray } from '~/utils/arrayHelper'

interface PostLinksFormProps {
  links: string[]
  setLinks: Dispatch<SetStateAction<string[]>>
  disabledLinks: boolean
}

const MAX_LINKS_ALLOWED = 3
export const regexLink = /^((?:https?:\/\/)?[^./]+(?:\.[^./]+)+(?:\/.*)?)$/g

const PostLinksForm = ({ links, setLinks, disabledLinks }: PostLinksFormProps) => {
  const handleChange = (text: string, index: number) => {
    const valueNew = replaceValueFromIndex(links, index, text)
    setLinks(valueNew)
  }

  const addLink = () => {
    const valueNew = addToArray(links, '')
    setLinks(valueNew)
  }

  const removeLink = (index: number) => {
    const newValues = removeFromArray(links, index)
    setLinks(newValues)
  }

  const checkRegex = (i: number) => {
    if (links[i] === '') return false
    const split = links[i].split('.')
    if (split.length === 2 && split[0] === 'www') return true
    const linkRegex = new RegExp(regexLink)
    const test = linkRegex.test(links[i])
    return !test
  }

  return (
    <View>
      {links.map((val, i) => (
        <View key={`link_${i + 1}`}>
          <Input
            value={links[i]}
            title='Add link'
            onChangeText={text => handleChange(text, i)}
            placeholder='Enter link'
            autoCapitalize='none'
            disabled={disabledLinks}
            error={checkRegex(i)}
            style={StyleSheet.flatten([styles.inputAddLink, disabledLinks && styles.disabled])}
          />
          <View style={styles.actionsContainer}>
            {i < MAX_LINKS_ALLOWED - 1 && i === links.length - 1 && (
              <>
                <Icon name='add-circle-icon' size={14} color={theme.darkGreenColor} />
                <Text
                  style={[styles.actionText, disabledLinks && styles.disabled]}
                  onPress={!disabledLinks ? addLink : () => null}
                >
                  Add More
                </Text>
              </>
            )}
            {links.length > 1 && (
              <>
                <Icon name='remove-circle-icon' size={14} color={theme.darkGreenColor} />
                <Text
                  style={[styles.actionText, disabledLinks && styles.disabled]}
                  onPress={!disabledLinks ? () => removeLink(i) : () => null}
                >
                  Remove
                </Text>
              </>
            )}
          </View>
        </View>
      ))}
    </View>
  )
}

export default PostLinksForm
