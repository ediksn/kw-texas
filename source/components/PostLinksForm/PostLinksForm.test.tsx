import React from 'react'
import { render, RenderAPI } from '@testing-library/react-native'
import { PostLinksForm } from '~/components'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

describe('Render Header Post Links Form', () => {
  let component: RenderAPI

  beforeEach(() => {
    component = render(<PostLinksForm setLinks={() => ({})} disabledLinks={false} links={[]} />)
  })

  it('Match with the snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('Renders correctly', () => {
    expect(component).toBeDefined()
  })
})
