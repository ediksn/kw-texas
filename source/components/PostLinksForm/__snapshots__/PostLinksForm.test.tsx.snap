// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Header Post Links Form Match with the snapshot 1`] = `<View />`;
